packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Expressions/NewInstanceExpression.cs

```diff
@@ -5,20 +5,30 @@
 
 namespace Microsoft.Generator.CSharp.Expressions
 {
-    public sealed record NewInstanceExpression(CSharpType Type, IReadOnlyList<ValueExpression> Parameters, ObjectInitializerExpression? InitExpression = null) : ValueExpression
+    public sealed record NewInstanceExpression(CSharpType? Type, IReadOnlyList<ValueExpression> Parameters, ObjectInitializerExpression? InitExpression = null) : ValueExpression
     {
         private const int SingleLineParameterThreshold = 6;
 
         internal override void Write(CodeWriter writer)
         {
-            writer.Append($"new {Type}");
-            if (Parameters.Count > 0 || InitExpression is not { Parameters.Count: > 0 })
+            writer.Append($"new");
+            if (Type is not null)
+                writer.Append($" {Type}");
+            if (Parameters.Count > 0 || InitExpression is not { Values.Count: > 0 })
             {
                 writer.WriteArguments(Parameters, Parameters.Count < SingleLineParameterThreshold);
             }
 
-            if (InitExpression is { Parameters.Count: > 0 })
+            if (InitExpression is not null)
             {
+                if (!InitExpression.UseSingleLine)
+                {
+                    writer.WriteLine();
+                }
+                else
+                {
+                    writer.AppendRaw(" ");
```

Comment:
nit: can we use `string.Empty` here?

Reply:
this isn't empty its a single space character.

Reply:
ah you're right. Ignore this comment then.